cli/flox/src/commands/gc.rs

```diff
@@ -1,23 +1,291 @@
-use anyhow::Result;
+use std::io::{BufRead, BufReader};
+use std::process::{Stdio, exit};
+
+use anyhow::{Context, Result, anyhow, bail};
 use bpaf::Bpaf;
 use flox_rust_sdk::flox::Flox;
 use flox_rust_sdk::models::env_registry;
-use tracing::instrument;
+use flox_rust_sdk::providers::nix::nix_base_command;
+use tracing::field::Empty;
+use tracing::{Span, debug, info_span, instrument};
 
 use crate::message;
 
 #[derive(Bpaf, Debug, Clone)]
 pub struct Gc {}
 
 impl Gc {
-    #[instrument(
-        skip_all,
-        fields(progress = "Garbage collecting unused environment data")
-    )]
+    #[instrument(skip_all, fields(progress = "Collecting garbage"))]
     pub fn handle(self, flox: Flox) -> Result<()> {
-        env_registry::garbage_collect(&flox)?;
+        {
+            let span = info_span!(
+                "prune_environments",
+                progress = "Garbage collecting stale environments"
+            );
+            let _ = span.enter();
+            env_registry::garbage_collect(&flox)?;
+        }
```

Comment:
nit: maybe factor this out jsut for a cleaner handle function

Reply:
Factor what out? This one function call or the span stuff on `garbage_collect`?